### ISSUE TITLE
Data structure for the leaderboard

### ISSUE THREAD

**Issue by JackKelly** [JackKelly/adventures-in-energy-forecasting#1]
## How to store data for each experiment?

- Maybe start with a public [wandb table](https://wandb.ai/site/articles/intro-to-mlops-machine-learning-experiment-tracking/#:~:text=to%20a%20step.-,3.%20Information%20Retrieval,-You%20can%20get).
- But this might not be very easy for other people to contribute to?
- Maybe a directory of JSON files, one JSON file per experiment? ([`polars.scan_ndjson`](https://docs.pola.rs/api/python/dev/reference/api/polars.scan_ndjson.html) can read a directory of JSON files)? Where each JSON file links to the JSON schema used for that experiment?

## Attributes to record for each experiment

These tags will inevitably change as the project evolves. Don't obsess too much about getting this right from the beginning. Instead, make a start with a "best effort" schema. And be ready to change the schema over time. It won't be _that_ much effort to automatically update

- Experiment ID
- Summary stats of the performance of the model on the validation dataset:
    - NMAE, NMBE, RMSE, and histogram of normalised errors for:
        - the entire validation dataset
        - each forecast horizon
        - each hour of day
- Links to code:
    - to the code (and git commit) for the code used to train the model
    - to the code (and git commit) for the code used to measure performance on the test set
- Model type (XGBoost vs TFT etc. Maybe include major architectural elements in larger models. e.g. CNN -> transformer.)
- Training time (measured in GPU hours? Or TFLOPs?)
- Link to the full output of the model during the final validation run (i.e. a parquet file of the predictions created by the model, in a standard format, stored on object storage)
- Is the code open source? (it should be)
- Is all the training data open source?
- Inputs:
    - Quantity of training data used (per input, for models that can handle missing inputs)
    - Links to all the data sources (so people can reproduce the experiment)
    - A flag for whether a re-analysis dataset was used at inference time
    - NWPs
        - Per provider:
            - Per variable:
                - Size (x * y * vertical levels)
    - Context window
        - Length (in hours)
        - Variables included in the context window (e.g. solar power just from the PV system we're predicting for; or solar power from all neighbouring PV systems; or satellite data; or NWP; etc.)
        - How is data selected for the context window? (e.g. is it simply the last 7 days. Or the "most similar" days?)
    - Physics-based PV forecast?
        - Using inferred or nominal tilt & orientation?
    - Satellite data
    - Ground-based weather data
        - Irradiance?
        - Temerature?
        - etc.
    -

**Comment by JackKelly** [comment]
Related OCF doc: https://docs.google.com/document/d/1tiBgGtwOMII9Kpd2Wlh2zDB7bMHLeysr1cCBghLPOrY/edit?tab=t.0
